OptimizelySDK/ProjectConfig.cs

```diff
@@ -353,5 +359,90 @@ public Variation GetVariationFromId(string experimentKey, string variationId)
+        /// <param name="userId">The User ID</param>
+        /// <returns>Variation entity on which the given user and experiment should be forced into.</returns>
+        public Variation GetForcedVariation(string experimentKey, string userId)
+        {
+            if (_ForcedVariationMap.ContainsKey(userId) == false)
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format(@"User ""{0}"" is not in the forced variation map.", userId));
+                return null;
+            }
+
+            Dictionary<string, string> experimentToVariationMap = _ForcedVariationMap[userId];
+
+            string experimentId = GetExperimentFromKey(experimentKey).Id;
+
+            // this case is logged in getExperimentFromKey
+            if (string.IsNullOrEmpty(experimentId))
+                return null;
+            
+            if (experimentToVariationMap.ContainsKey(experimentId) == false)
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format(@"No experiment ""{0}"" mapped to user ""{1}"" in the forced variation map.", experimentKey, userId));
+                return null;
+            }
+            
+            string variationId = experimentToVariationMap[experimentId];
+
+            if (string.IsNullOrEmpty(variationId))
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format(@"No variation mapped to experiment ""{0}"" in the forced variation map.", experimentKey));
+                return null;
+            }
+            
+            string variationKey = GetVariationFromId(experimentKey, variationId).Key;
+
+            // this case is logged in getVariationFromKey
+            if (string.IsNullOrEmpty(variationKey))
+                return null;
+
+            Logger.Log(LogLevel.DEBUG, string.Format(@"Variation ""{0}"" is mapped to experiment ""{1}"" and user ""{2}"" in the forced variation map", variationKey, experimentKey, userId));
+            
+            Variation variation = GetVariationFromKey(experimentKey, variationKey);
+
+            return variation;
+        }
+        
+        /// <summary>
+        /// Sets an associative array of user IDs to an associative array of experiments to forced variations.
+        /// </summary>
+        /// <param name="experimentKey">key for Experiment</param>
+        /// <param name="userId">The User ID</param>
+        /// <param name="variationKey">key for Variation</param>
+        /// <returns>A boolean value that indicates if the set completed successfully.</returns>
+        public bool SetForcedVariation(string experimentKey, string userId, string variationKey)
+        {
+            var experimentId = GetExperimentFromKey(experimentKey).Id;
+
+            // this case is logged in getExperimentFromKey
+            if (string.IsNullOrEmpty(experimentId))
+                return false;
+
+            if (string.IsNullOrEmpty(variationKey))
+            {
+                _ForcedVariationMap.Remove(userId);
```

Comment:
You don't want to remove all the experiment-to-variation mapping for that user ID. You should only be removing the forced variation mapping for the specified experiment and user.